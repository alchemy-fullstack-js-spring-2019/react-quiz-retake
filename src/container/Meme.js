import React, { PureComponent } from 'react';
import TextSelector from '../components/TextSelector';
import Display from '../components/Display';

export default class Meme extends PureComponent {
  state = {
    img: '',
    headerText: '',
    footerText: ''
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render () {
    const { img, headerText, footerText } = this.state;
    return (
      <>
        <TextSelector
          img={img}
          headerText={headerText}
          footerText={footerText}
          handleChange={this.handleChange}
        />
        <Display
          img={img}
          headerText={headerText}
          footerText={footerText}
        />
      </>
    );
  }
}
