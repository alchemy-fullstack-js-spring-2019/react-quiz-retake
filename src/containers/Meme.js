import React, { PureComponent } from 'react';
import Display from '../components/Display';
import TextSelector from '../components/TextSelector';

export default class Meme extends PureComponent {
  state = {
    headerText: 'Make a',
    footerText: 'cool meme!',
    image: 'https://njmom.com/wp-content/uploads/2019/01/iStock-532529153.jpg'
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { headerText, footerText, image } = this.state;
  
    return (
      <>
      <TextSelector headerText={headerText} footerText={footerText} image={image} handleChange={this.handleChange}/>
      <Display headerText={headerText} footerText={footerText} image={image}/>
      </>
    );
  }
}
