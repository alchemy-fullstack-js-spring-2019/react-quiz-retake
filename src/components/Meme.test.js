import React from 'react';
import { shallow } from 'enzyme';
import Meme from './Meme';

describe('Meme component', () => {
  it('renders a Meme', () => {
    const wrapper = shallow(<Meme />);
    expect(wrapper).toMatchSnapshot();
  });
});
