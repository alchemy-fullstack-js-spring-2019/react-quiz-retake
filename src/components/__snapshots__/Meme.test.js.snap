// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meme component renders a Meme 1`] = `
<Fragment>
  <TextSelector
    footerText=""
    handleChange={[Function]}
    headerText=""
    image=""
  />
  <Display
    footerText=""
    headerText=""
    image=""
  />
</Fragment>
`;
